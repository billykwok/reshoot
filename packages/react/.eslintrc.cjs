module.exports = {
  root: false,
  env: {
    browser: true,
    node: true,
    commonjs: true,
    es6: true,
    'jest/globals': true,
  },
  parser: '@babel/eslint-parser',
  parserOptions: {
    ecmaVersion: 'latest',
    sourceType: 'module',
    ecmaFeatures: { impliedStrict: true, jsx: true },
    babelOptions: { configFile: './babel.config.cjs' },
  },
  settings: {
    react: { version: 'detect' },
    'import/extensions': ['.js', '.jsx', '.json'],
    'import/resolver': { node: { moduleDirectory: ['node_modules'] } },
  },
  plugins: [
    'import',
    'prettier',
    'react',
    'react-hooks',
    'jsx-a11y',
    'testing-library',
    'jest',
  ],
  extends: [
    'eslint:recommended',
    'plugin:react/recommended',
    'plugin:jsx-a11y/recommended',
    'plugin:prettier/recommended',
    'prettier',
    'plugin:testing-library/react',
    'plugin:jest/recommended',
  ],
  rules: {
    'react/prop-types': 'off',
    'react/jsx-uses-react': 'off',
    'react/react-in-jsx-scope': 'off',
    'jsx-a11y/image-has-content': 'off',
    'jsx-a11y/heading-has-content': 'off',
    'jsx-a11y/anchor-has-content': 'off',
    'react-hooks/rules-of-hooks': 'error',
    'prefer-spread': 'off',
    'consistent-return': 'off',
    'max-len': 'off',
    'no-console': 'off',
  },
  overrides: [
    {
      files: ['*.ts', '*.tsx'],
      parser: '@typescript-eslint/parser',
      parserOptions: {
        ecmaVersion: 'latest',
        sourceType: 'module',
        ecmaFeatures: { impliedStrict: true, jsx: true },
        project: ['./tsconfig.json'],
      },
      settings: {
        react: { version: 'detect' },
        'import/extensions': ['.ts', '.tsx'],
        'import/resolver': { node: { moduleDirectory: ['node_modules'] } },
        'import/parsers': { '@typescript-eslint/parser': ['.ts', '.tsx'] },
      },
      plugins: [
        '@typescript-eslint',
        'import',
        'prettier',
        'react',
        'react-hooks',
        'jsx-a11y',
        'testing-library',
        'jest',
      ],
      extends: [
        'eslint:recommended',
        'plugin:@typescript-eslint/eslint-recommended',
        'plugin:@typescript-eslint/recommended',
        'plugin:@typescript-eslint/recommended-requiring-type-checking',
        'plugin:react/recommended',
        'plugin:jsx-a11y/recommended',
        'plugin:prettier/recommended',
        'prettier',
        'plugin:testing-library/react',
        'plugin:jest/recommended',
      ],
      rules: {
        '@typescript-eslint/no-var-requires': 'off',
        '@typescript-eslint/no-explicit-any': 'off',
        '@typescript-eslint/no-floating-promises': 'off',
        '@typescript-eslint/explicit-function-return-type': 'off',
        '@typescript-eslint/explicit-module-boundary-types': [
          'error',
          { allowArgumentsExplicitlyTypedAsAny: true },
        ],
        '@typescript-eslint/prefer-includes': 'off',
        'react/prop-types': 'off',
        'react/jsx-uses-react': 'off',
        'react/react-in-jsx-scope': 'off',
        'jsx-a11y/image-has-content': 'off',
        'jsx-a11y/heading-has-content': 'off',
        'jsx-a11y/anchor-has-content': 'off',
        'react-hooks/rules-of-hooks': 'error',
      },
    },
  ],
};
