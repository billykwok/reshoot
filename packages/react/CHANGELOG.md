# @reshoot/react

## 2.0.5

### Patch Changes

- a0288f4: fix: support hash character in inline options

## 2.0.4

### Patch Changes

- ec76336: fix: improve rollup setup and update dependencies
- Updated dependencies [ec76336]
  - @reshoot/types@2.0.3

## 2.0.3

### Patch Changes

- 0e54ec5: fix: fix style exports

## 2.0.2

### Patch Changes

- 510a7da: fix: fix incorrect extensions in package.json

## 2.0.1

### Patch Changes

- 226a185: fix: add node version restriction and fix rollup config
- Updated dependencies [226a185]
  - @reshoot/types@2.0.1

## 2.0.0

### Major Changes

- 8d162cb: Introducing Reshoot 2

  Reshoot has been rewritten from ground up to support more use cases. Please refer to README.md for details.

### Patch Changes

- Updated dependencies [8d162cb]
  - @reshoot/types@2.0.0
