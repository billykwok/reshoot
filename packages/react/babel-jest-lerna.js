module.exports = require('babel-jest').createTransformer({
  rootMode: 'upward',
  presets: [
    ['@babel/preset-env', { targets: { node: 'current' } }],
    [
      '@babel/preset-typescript',
      {
        isTSX: true,
        allExtensions: true,
        allowNamespaces: true,
        allowDeclareFields: true,
        onlyRemoveTypeImports: true,
      },
    ],
    ['@babel/preset-react', { runtime: 'automatic', useBuiltIns: true }],
    'linaria/babel',
  ],
  plugins: [
    [
      '@babel/plugin-transform-runtime',
      {
        corejs: { version: 3, proposals: true },
        helpers: true,
        regenerator: true,
      },
    ],
    'dynamic-import-node',
    ['const-enum', { transform: 'constObject' }],
  ],
});
