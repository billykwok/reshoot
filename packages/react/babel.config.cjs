module.exports = {
  targets: { esmodules: true },
  presets: [
    ['@babel/preset-env', { targets: { esmodules: true }, bugfixes: true }],
    ['@babel/preset-typescript', { isTSX: true, allExtensions: true }],
    ['@babel/preset-react', { runtime: 'automatic', useBuiltIns: true }],
    '@linaria',
  ],
  plugins: [
    [
      '@babel/plugin-transform-runtime',
      { corejs: { version: 3 }, helpers: true },
    ],
  ],
  compact: true,
  minified: true,
};
