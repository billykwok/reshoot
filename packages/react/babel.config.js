module.exports = {
  presets: [
    ['@babel/preset-env', { targets: { esmodules: true }, bugfixes: true }],
    [
      '@babel/preset-typescript',
      {
        isTSX: true,
        allExtensions: true,
        allowNamespaces: true,
        allowDeclareFields: true,
        onlyRemoveTypeImports: true,
      },
    ],
    ['@babel/preset-react', { runtime: 'automatic', useBuiltIns: true }],
    'linaria/babel',
  ],
  plugins: [
    [
      '@babel/plugin-transform-runtime',
      { corejs: { version: 3, proposals: true }, helpers: true },
    ],
    ['const-enum', { transform: 'constObject' }],
  ],
};
