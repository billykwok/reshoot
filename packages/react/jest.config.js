module.exports = {
  displayName: '@reshoot/react',
  snapshotResolver: '<rootDir>/test/snapshotResolver.js',
  setupFilesAfterEnv: ['<rootDir>/test/setup.js'],
  coverageProvider: 'v8',
  verbose: true,
  transform: { '^.+\\.[tj]sx?$': './babel-jest-lerna.js' },
};
