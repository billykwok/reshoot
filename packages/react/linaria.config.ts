export default {
  evaluate: true,
  displayName: process.env.NODE_ENV === 'development',
};
