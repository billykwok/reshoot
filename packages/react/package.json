{
  "name": "@reshoot/react",
  "description": "A React component that lazily renders responsive images with LQIPs",
  "version": "1.6.4",
  "type": "commonjs",
  "exports": {
    "import": "./lib/es/index.js",
    "require": "./lib/index.js"
  },
  "main": "./lib/index.js",
  "types": "./types/index.d.ts",
  "files": [
    "lib",
    "types"
  ],
  "sideEffects": [
    "*.css"
  ],
  "publishConfig": {
    "access": "public"
  },
  "author": "Billy Kwok <opensource@billykwok.me>",
  "license": "MIT",
  "keywords": [
    "reshoot",
    "srcset",
    "image",
    "img",
    "react",
    "responsive",
    "babel"
  ],
  "repository": "https://github.com/billykwok/reshoot",
  "bugs": {
    "url": "https://github.com/billykwok/reshoot/issues"
  },
  "homepage": "https://github.com/billykwok/reshoot/tree/master/packages/react",
  "scripts": {
    "lint": "eslint ./src/**",
    "clean": "rm -rf lib types",
    "build": "yarn clean && rollup -c rollup.config.js && rollup -c rollup.config.js --format es --dir ./lib/es && tsc --project tsconfig.json"
  },
  "peerDependencies": {
    "@babel/runtime-corejs3": "^7.11.2",
    "object-assign": "4.1.1",
    "react": "^16.13.1"
  }
}
