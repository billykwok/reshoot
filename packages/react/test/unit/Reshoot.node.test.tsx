import { describe, beforeAll, afterEach, test, expect } from '@jest/globals';
import * as React from 'react';
import { renderToString } from 'react-dom/server';

describe('Reshoot', () => {
  const baseData = {
    sources: [
      {
        type: 'image/webp',
        srcSet: 'image-480.webp 480w, image-640.webp 640w',
      },
    ],
    src: 'image.jpg',
    width: 100,
    height: 80,
    color: '#fff',
  };

  beforeAll(() => {
    jest.doMock('../../src/utils/isBrowser', () => ({
      __esModule: true,
      default: false,
    }));
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  test('server side rendering', async () => {
    const { Reshoot } = await import('../../src/Reshoot');
    const ref = React.createRef<HTMLImageElement>();
    const html = renderToString(
      <Reshoot
        data-testid="123"
        data={{ ...baseData, aspectRatio: 2 }}
        imgProps={{ alt: 'Test image' }}
        ref={ref}
      />
    );
    expect(html).toMatchSnapshot();
  });
});
