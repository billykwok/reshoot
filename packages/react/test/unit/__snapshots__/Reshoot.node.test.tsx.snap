// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reshoot server side rendering 1`] = `"<div class=\\"ax4jui0\\" style=\\"color:#fff;--r:2\\" data-testid=\\"123\\"><picture><source type=\\"image/webp\\" srcSet=\\"image-480.webp 480w, image-640.webp 640w\\"/><img src=\\"image.jpg\\" alt=\\"Test image\\" width=\\"100\\" height=\\"80\\" crossorigin=\\"\\"/></picture></div>"`;
